### ISSUE TITLE
fix(solvers): `zero_mean` semantics — spherical Helmholtz zeroes l=0 when α>0; Fourier `alpha=0, zero_mean=False` returns −mean(f); DCT-I "zero mean" is the trapezoid mean

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/spectraldiffx#92]
Part of #80.

Three related defects in how the null mode of Poisson/Helmholtz solves is handled. All are reproduced; fix them together because the resolution is one consistent policy.

## A. `SphericalHelmholtzSolver.solve` zeroes the mean when α > 0

`spectraldiffx/_src/spherical/solvers.py:176-179` (docstring: `zero_mean` is "ignored effectively when α > 0") vs `:195-206` (code applies `where(l == 0, 0, phi_hat)` whenever `zero_mean`, default `True`).

```python
import jax; jax.config.update("jax_enable_x64", True)
import jax.numpy as jnp, numpy as np
from spectraldiffx import SphericalGrid2D, SphericalHelmholtzSolver, SphericalDerivative2D
g = SphericalGrid2D.from_N_L(Nx=32, Ny=16, Lx=2*np.pi, Ly=np.pi)
d = SphericalDerivative2D(g); s = SphericalHelmholtzSolver(g)
th = g.y[:, None]; ph = g.x[None, :]
phi_true = 2.0 + jnp.sin(th)**2 * jnp.cos(2*ph)      # mean 2
alpha = 0.7
f = d.laplacian(phi_true) - alpha * phi_true
print(float(jnp.abs(s.solve(f, alpha=alpha) - phi_true).max()))                   # 2.00e+00
print(float(jnp.abs(s.solve(f, alpha=alpha, zero_mean=False) - phi_true).max()))  # 2.1e-13
```

Fix: `zero_mean: bool | None = None` meaning "zero the null mode iff α == 0"; when α is a Python scalar this is static; when traced, apply `jnp.where(alpha == 0, ...)`.

## B. Fourier `SpectralHelmholtzSolver*.solve(alpha=0, zero_mean=False)` returns `mean(φ) = −mean(f)`

`spectraldiffx/_src/fourier/solvers.py:2045-2050` and the 2D/3D equivalents replace the zero denominator by 1, so the undefined mode becomes `−f̂₀`. With `mean(f) = 2` the solution comes back with `mean(φ) = −2`. The same happens in `SphericalPoissonSolver.solve(zero_mean=False)` (`spherical/solvers.py:118-127`: `f = 1 + cos θ` → `mean(φ) = −1.0`).

Fix: when α == 0 the null mode is undefined; either always project it out (ignore `zero_mean`) or raise if `zero_mean=False` and α == 0. Do not return an arbitrary value.

## C. DCT-I (regular Neumann) "zero-mean gauge" is the trapezoid-weighted mean, and an incompatible RHS is silently projected

`solve_helmholtz_dct1_*` (`solvers.py:1028, 1506, 1834`), `RegularNeumannHelmholtzSolver2D`, and any `"neumann"` axis of `solve_helmholtz_2d/3d`. The DCT-I FD2 matrix is not symmetric; its left null vector is `w = (1, 2, …, 2, 1)` (trapezoid weights). Solvability is `w·f = 0` and the gauge the solver actually imposes is `w·ψ = 0`. Docstrings say "the domain mean of ψ is zero"; the tests assert `|mean(ψ)| < 1e-6` and pass only because the chosen eigenfunction is symmetric.

```
plain-zero-mean rhs:      |Aψ − f| = 1.35e-01,  mean(ψ) = −3.0e-03,  trapz-mean(ψ) = 6e-18
trapezoid-zero-mean rhs:  |Aψ − f| = 4.7e-16
```

(Repro: 1D `solve_helmholtz_dct1_1d` on N = 16 with `f = A @ x_true` for a random `x_true`, then subtract either `mean(f)` or `(w·f)/(w·1)`.) Same in 2D and for periodic×neumann.

Fix: document the actual gauge (`w·ψ = 0`); optionally emit a warning when `|w·f| / |f|` is not negligible, since every null-mode solver silently solves for `f − proj(f)`. The tests for DCT-I should assert `w·ψ = 0` and use an asymmetric RHS.

## Acceptance criteria

- [ ] Spherical Helmholtz: default call with α > 0 recovers a field with non-zero mean to 1e-12.
- [ ] Fourier and spherical Poisson: `zero_mean=False` with α = 0 either projects or raises; never returns `−mean(f)`.
- [ ] DCT-I tests assert the trapezoid gauge with an asymmetric RHS; docstrings corrected.
- [ ] One policy documented in `docs/theory/elliptic_solvers.md`: what happens to the null mode, for each BC family.
